stdlib/Dates/src/deprecated.jl

```diff
@@ -68,3 +67,21 @@ end
 
 @deprecate argerror(msg::String) ArgumentError(msg) false
 @deprecate argerror() nothing false
+
+import Base: ceil, round, floor
+
+Base.@deprecate(round(dt::TimeType,  p::Period, r::RoundingMode{:NearestTiesUp}), round(p, dt, r))
+
+Base.@deprecate(round(x::TimeType, p::Period), round(p, x))
```

Comment:
I like that you defined this separately from the three argument version to give better warning messages.